pkg/cmd/root/extension.go

```diff
@@ -7,40 +7,36 @@ import (
 	"strings"
 	"time"
 
+	"github.com/cli/cli/v2/internal/update"
 	"github.com/cli/cli/v2/pkg/extensions"
 	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/cli/cli/v2/utils"
 	"github.com/spf13/cobra"
 )
 
 type ExternalCommandExitError struct {
 	*exec.ExitError
 }
 
-type extensionReleaseInfo struct {
-	CurrentVersion string
-	LatestVersion  string
-	Pinned         bool
-	URL            string
-}
-
-func NewCmdExtension(io *iostreams.IOStreams, em extensions.ExtensionManager, ext extensions.Extension) *cobra.Command {
-	updateMessageChan := make(chan *extensionReleaseInfo)
+func NewCmdExtension(io *iostreams.IOStreams, em extensions.ExtensionManager, ext extensions.Extension, checkExtensionReleaseInfo func(extensions.ExtensionManager, extensions.Extension) (*update.ReleaseInfo, error)) *cobra.Command {
```

Comment:
Open to ideas on alternatives to `checkExtensionReleaseInfo` function approach to make this testable. 🤷